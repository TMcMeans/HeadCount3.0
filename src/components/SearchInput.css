@import url('https://fonts.googleapis.com/css?family=Poppins');

input[type='text'] {
  width: 300px;
  min-width: 320px;
  height: 30px;
  padding: 10px 10px 10px 5px;
  border-bottom: 1px solid #4a314d;
  font-size: 1.5em;
  font-family: 'Poppins', sans-serif;
  border-top: none;
  border-right: none;
  border-left: none;
}

input[type='text']:focus {
  outline: 0.3px solid orange;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  box-shadow: inset 0 0 0 4px #f0af4d, 0 0 1px rgba(0, 0, 0, 0);
  border-bottom: 1px solid white;
}
