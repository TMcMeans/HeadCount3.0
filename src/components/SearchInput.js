import React, { Component } from 'react';

import './SearchInput.css';

class SearchInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchInput: ''
    };
  }

  handleInputChange = e => {
    this.setState({
      searchInput: e.target.value
    });
    this.props.filterCards(this.state.searchInput);
  };

  render() {
    const { searchInput } = this.state;
    return (
      <form>
        <input
          type="text"
          name="searchInput"
          value={searchInput}
          onChange={this.handleInputChange}
          placeholder="Search"
        />
      </form>
    );
  }
}

export default SearchInput;
