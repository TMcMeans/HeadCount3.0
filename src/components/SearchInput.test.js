import React from 'react';
import { shallow, mount } from 'enzyme';

import SearchInput from './SearchInput';

describe('SearchInput', () => {
  it('should match snapshot with all html elements rendered with correct data', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<SearchInput filterCards={mockFunc} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('updates state of the searchInput with user input values', () => {
    const mockFunc = jest.fn();

    const wrapper = mount(<SearchInput filterCards={mockFunc} />);

    const mockEvent = { target: { value: 'colorado', name: 'searchInput' } };

    const expectedState = {
      searchInput: 'colorado'
    };

    wrapper.instance().handleInputChange(mockEvent);

    expect(wrapper.state()).toEqual(expectedState);
  });
});
