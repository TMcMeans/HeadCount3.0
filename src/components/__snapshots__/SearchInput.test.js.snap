// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should match snapshot with all html elements rendered with correct data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput
    filterCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        name="searchInput"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "name": "searchInput",
        "onChange": [Function],
        "placeholder": "Search",
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          name="searchInput"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "searchInput",
          "onChange": [Function],
          "placeholder": "Search",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
